### ISSUE TITLE
ClassMemberImpliedModifierCheck: nested enum/interface missing implied static not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20227]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/modifier/classmemberimpliedmodifier.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassMemberImpliedModifier"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputClassMemberImpliedModifierCompact.java <<'EOF'
enum Color { RED, GREEN, BLUE }
interface Printable { void print(); }
void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputClassMemberImpliedModifierCompact.java

Starting audit...
Audit done.
```

Same code inside an ordinary class:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > OrdinaryClassMemberImpliedModifier.java <<'EOF'
public class OrdinaryClassMemberImpliedModifier {
    enum Color { RED, GREEN, BLUE }
    interface Printable { void print(); }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml OrdinaryClassMemberImpliedModifier.java

Starting audit...
[ERROR] OrdinaryClassMemberImpliedModifier.java:2:5: Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
[ERROR] OrdinaryClassMemberImpliedModifier.java:3:5: Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
Audit done.
Checkstyle ends with 2 errors.
```

---

**Describe what you expect in detail.**

ClassMemberImpliedModifierCheck must flag a nested `enum` or `interface` declared at the top level of a JEP 512 compact source file as missing the implied `static` modifier. The check gates on `ScopeUtil.isInClassBlock(ast)`; for a nested type whose enclosing class is the implicit class (not represented by a `CLASS_DEF`), this returns false and the check silently skips the declaration. The nested types are still implicitly `static` per JLS, so the missing modifier should be flagged.

Expected on the compact input: 2 violations (`Implied modifier 'static' should be explicit.` on `Color` and on `Printable`), identical to the ordinary class above.

**Comment by stoyanK7** [comment]
Thought: I wonder if this would bother people. On one hand, yes,  _technically_ this is what the check should do since the compact source file is a static class. On the other hand, how many users would know that (the compact source file is a static class) and expect such behaviour?

I think it will be good to mention that in the docs.

**Comment by vivek-0509** [comment]
@stoyanK7 Yes, I'll add a note to the check's documentation. I'll include a short paragraph (with a compact-source-file example) explaining that top-level nested enums/interfaces/records in a JEP 512 compact source file are flagged because the implicit class makes them static, just like nested enums, interfaces, and records in an ordinary class

**Comment by surajgojanur** [comment]
I’d like to work on this. I’ll add compact-source coverage and fix the implied `static` handling for top-level enum/interface declarations.

**Comment by romani** [comment]
Fix is merged
